test-utils/service-base-test-utils/src/main/java/org/eclipse/hono/service/test/ProtocolAdapterTestSupport.java

```diff
@@ -0,0 +1,264 @@
+        when(commandConsumerFactory.connect()).thenReturn(Future.succeededFuture(mock(HonoConnection.class)));
+        when(commandConsumerFactory.isConnected()).thenReturn(Future.succeededFuture());
+
+        commandTargetMapper = mock(CommandTargetMapper.class);
+
+        credentialsClientFactory = mock(CredentialsClientFactory.class);
+        when(credentialsClientFactory.connect()).thenReturn(Future.succeededFuture(mock(HonoConnection.class)));
+        when(credentialsClientFactory.isConnected()).thenReturn(Future.succeededFuture());
+
+        deviceConnectionClientFactory = mock(DeviceConnectionClientFactory.class);
+        when(deviceConnectionClientFactory.connect()).thenReturn(Future.succeededFuture(mock(HonoConnection.class)));
+        when(deviceConnectionClientFactory.isConnected()).thenReturn(Future.succeededFuture());
+
+        downstreamSenderFactory = mock(DownstreamSenderFactory.class);
+        when(downstreamSenderFactory.connect()).thenReturn(Future.succeededFuture(mock(HonoConnection.class)));
+        when(downstreamSenderFactory.isConnected()).thenReturn(Future.succeededFuture());
+
+        registrationClientFactory = mock(RegistrationClientFactory.class);
+        when(registrationClientFactory.connect()).thenReturn(Future.succeededFuture(mock(HonoConnection.class)));
+        when(registrationClientFactory.isConnected()).thenReturn(Future.succeededFuture());
+
+        registrationClient = mock(RegistrationClient.class);
+        when(registrationClientFactory.getOrCreateRegistrationClient(anyString())).thenReturn(Future.succeededFuture(registrationClient));
+
+        tenantClientFactory = mock(TenantClientFactory.class);
+        when(tenantClientFactory.connect()).thenReturn(Future.succeededFuture(mock(HonoConnection.class)));
+        when(tenantClientFactory.isConnected()).thenReturn(Future.succeededFuture());
+
+        connectionEventProducerContext = mock(ConnectionEventProducer.Context.class);
+        when(connectionEventProducerContext.getMessageSenderClient()).thenReturn(downstreamSenderFactory);
+        when(connectionEventProducerContext.getTenantClientFactory()).thenReturn(tenantClientFactory);
+
+        properties = givenDefaultConfigurationProperties();
+    }
+
+    /**
+     * Creates default configuration for the adapter.
+     *
+     * @return The configuration properties.
+     */
+    protected abstract C givenDefaultConfigurationProperties();
+
+    /**
+     * Sets up a new adapter instance as the unit under test.
+     * <p>
+     * {@linkplain #newAdapter(ProtocolAdapterProperties) Creates a new instance}
+     * and then {@linkplain #setCollaborators(AbstractProtocolAdapterBase)
+     * sets all collaborators} on the new instance.
+     *
+     * @param configuration The configuration properties to apply.
+     * @return The adapter instance to test.
+     */
+    protected abstract T givenAnAdapter(C configuration);
```

Comment:
The javadoc (including the now unresolvable `#newAdapter` link) doesn't fit anymore.

In general the method doesn't seem to be needed in this class.

Reply:
true, I have removed the method ...